File: voe-replacement.py
import json
import requests

# API Key - https://voe.sx/settings (Account -> API Details)
voe_api_key = ''

# MySQL Database Configuration
database_username = 'wordpress'
database_password = ''
database_host = 'localhost'
database_database = 'wordpress'

# MySQL tables & their field that should be checked.
#  the script looks up: voe.sx/e/(code) and voe.sx/(code) and i.voe.sx/cache/(code)_storyboard
replacements = [
    {'table': 'wp_posts', 'field': 'post_content'},
    {'table': 'wp_postmeta', 'field': 'meta_value'},
    # ...
]

# Database maximal updates per link / limits the damage in case of errors.
database_max_updates_per_link = 50

# MySQL Connection (equivalent to $mysqli = new mysqli(...))
import pymysql.cursors
connection = pymysql.connect(
    host=database_host,
    user=database_username,
    password=database_password,
    db=database_database,
    cursorclass=pymysql.cursors.DictCursor
)

def replace_old_link_with_new_one(old_link, new_link):
    global replacements, connection, database_max_updates_per_link
    print(f" - replace VOE {old_link} with new one {new_link}\n")

    with connection.cursor() as cursor:
        for replacement in replacements:
            query = f"UPDATE {replacement['table']} SET {replacement['field']} = REPLACE({replacement['field']}, %s, %s) WHERE INSTR({replacement['field']}, %s) > 0 LIMIT {database_max_updates_per_link}"
            cursor.execute(query, (old_link, new_link, old_link))

    connection.commit()
    print("- Link has been successfully replaced.\n")

print("# VOE Deleted Files/DMCA Replacement #\n")

print("- get VOE Deleted/DMCA List\n")
dmca_url = f"https://voe.sx/api/dmca/list?key={voe_api_key}&pending=true"
r = requests.get(dmca_url)
data = json.loads(r.text)

DMCA_links = data['result']
server_time = data['server_time']

if DMCA_links:
    for DMCA_link in DMCA_links:
        print(f" - process {DMCA_link['file_code']}\n")

        del_time = DMCA_link['del_time']
        file_code = DMCA_link['file_code']

        if del_time >= server_time:
            link_found = False

            with connection.cursor() as cursor:
                for replacement in replacements:
                    if link_found:
                        continue

                    query = f"SELECT COUNT(*) as counter FROM {replacement['table']} WHERE INSTR({replacement['field']}, %s) > 0 LIMIT 1"
                    cursor.execute(query, (file_code,))
                    row = cursor.fetchone()
                    if row['counter'] > 0:
                        link_found = True

            if link_found:
                print("\n - clone VOE code {file_code}\n")
                r = requests.get(f"https://voe.sx/api/file/clone?key={voe_api_key}&file_code={file_code}")
                data = json.loads(r.text)
                new_link = data['result']['filecode']

                replace_old_link_with_new_one(f"voe.sx/e/{file_code}", f"voe.sx/e/{new_link}")
                replace_old_link_with_new_one(f"voe.sx/{file_code}", f"voe.sx/{new_link}")
                replace_old_link_with_new_one(f"i.voe.sx/cache/{file_code}_storyboard", f"i.voe.sx/cache/{new_link}_storyboard")
                print("\n")
            else:
                print("\n")
        else:
            print("\n")
    print("# completed, all reports processed. #\n")
else:
    print("No DMCA Links found.\n")

connection.close()

File: voe-replacement.php
<?php

// API Key - https://voe.sx/settings (Account -> API Details)
$voe_api_key = '';

// MySQL Database Configuration
$database_username = 'wordpress';
$database_password = '';
$database_host = 'localhost';
$database_database = 'wordpress';

// MySQL tables & their field that should be checked.
//  the script looks up: voe.sx/e/(code) and voe.sx/(code) and i.voe.sx/cache/(code)_storyboard
$replacements = [
    ['table' => 'wp_posts', 'field' => 'post_content'],
    ['table' => 'wp_postmeta', 'field' => 'meta_value'],
    // ...
];

// Database maximal updates per link / limits the damage in case of errors.
$database_max_updates_per_link = 50;

$mysqli = new mysqli($database_host, $database_username, $database_password, $database_database);
if ($mysqli->connect_errno) {
    die('Failed to connect to MySQL: ' . $mysqli->connect_error);
}

function replaceOldLinkWithNewOne($old_link, $new_link)
{
    global $replacements, $mysqli, $database_max_updates_per_link;
    echo " - replace VOE $old_link with new one $new_link" . "\r\n <br>";

    foreach ($replacements as $replacement) {
        $query = "UPDATE " . $replacement['table'] . " SET " . $replacement['field'] . " = REPLACE(" . $replacement['field'] . ", '$old_link', '$new_link') WHERE INSTR(" . $replacement['field'] . ", '$old_link') > 0 LIMIT $database_max_updates_per_link;";
        $mysqli->query($query);
    }
    echo "- Link has been successfully replaced." . "\r\n <br>";
}

echo "# VOE Deleted Files/DMCA Replacement #" . "\r\n <br>";
echo "\r\n <br>";

echo "- get VOE Deleted/DMCA List" . "\r\n <br>";
$dmca_url = "https://voe.sx/api/dmca/list?key=$voe_api_key&pending=true";
$r = file_get_contents($dmca_url);
$data = json_decode($r, true);

$DMCA_links = $data['result'];
$server_time = $data['server_time'];

if ($DMCA_links) {
    foreach ($DMCA_links as $DMCA_link) {
        echo " - process {$DMCA_link['file_code']}" . "\r\n <br>";

        $del_time = $DMCA_link['del_time'];
        $file_code = $DMCA_link['file_code'];

        if (strtotime($del_time) >= strtotime($server_time)) {
            $link_found = false;

            foreach ($replacements as $replacement) {
                if ($link_found) {
                    continue;
                }

                $cur = $mysqli->query("SELECT COUNT(*) as counter FROM " . $replacement['table'] . " WHERE INSTR(" . $replacement['field'] . ", '$file_code') > 0 LIMIT 1;");
                $row = $cur->fetch_assoc();
                if ($row['counter'] > 0) {
                    $link_found = true;
                }
            }

            if ($link_found) {
                echo "\r\n <br>";
                echo " - clone VOE code $file_code" . "\r\n <br>";
                $r = file_get_contents("https://voe.sx/api/file/clone?key=$voe_api_key&file_code=$file_code");
                $data = json_decode($r, true);
                $new_link = $data['result']['filecode'];

                replaceOldLinkWithNewOne("voe.sx/e/$file_code", "voe.sx/e/$new_link");
                replaceOldLinkWithNewOne("voe.sx/$file_code", "voe.sx/$new_link");
                replaceOldLinkWithNewOne('i.voe.sx/cache/' . $file_code . '_storyboard', 'i.voe.sx/cache/' . $new_link . '_storyboard');
                echo "\r\n <br>";
            } else {
                echo "\r\n <br>";
            }
        } else {
            echo "\r\n <br>";
        }
    }
    echo "# completed, all reports processed. #" . "\r\n <br>";
} else {
    echo "No DMCA Links found." . "\r\n <br>";
}

$mysqli->close();